### PR TITLE
Proper handling of boolean values that are not a flag

### PR DESCRIPTION
**Pull request by blankclemens**
..which have "true" as their default value and require a text parameter.

### CODE REVIEW

File: tools/openms/CompNovo.xml [hunk 1]
```diff
@@ -41,6 +41,11 @@
 #end if
 #if $param_algorithm_estimate_precursor_mz:
   -algorithm:estimate_precursor_mz
+  #if " " in str($param_algorithm_estimate_precursor_mz):
```
Comment (bgruening) [inline tools/openms/CompNovo.xml]: The only difference between if/else are the "quotes"?
The if condition can also be never true anymore? Is this all due to the converter?
Without looking much further I think you can remove the entire if and write `-algorithm:estimate_precursor_mz $param_algorithm_estimate_precursor_mz`

Comment (blankclemens) [inline tools/openms/CompNovo.xml]: You're right, I have this on my list.

Comment (blankclemens) [inline tools/openms/CompNovo.xml]: Fixed.

Comment (blankclemens) [inline tools/openms/CompNovo.xml]: Eh nope, forgot the command :)

Comment (blankclemens) [inline tools/openms/CompNovo.xml]: Now.

**Comment by bgruening** [comment]
Nice! 500 lines less code :)

**Pull request merged**